### PR TITLE
Direction finding connectionless locator sample app: add support for the nRF52820

### PR DESCRIPTION
**Pull request by ppryga**
Add required configuration and DTS overlay for support the nRF52820 SOC.

This PR should be merged after: https://github.com/nrfconnect/sdk-nrf/pull/4773
Also upmerge of sdk-zephyr repository has to be finished.

### CODE REVIEW

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst [hunk 1]
```diff
@@ -17,6 +17,7 @@ The sample supports the following development kits:
 .. table-from-rows:: /includes/sample_board_rows.txt
    :header: heading
    :rows: nrf52833dk_nrf52833
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/README.rst:19]: `:rows: nrf52833dk_nrf52833, nrf52833dk_nrf52820`

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CONFIG_BT_CTLR=y
+CONFIG_BT_LL_SW_SPLIT=y
+
+CONFIG_BT_CTLR_ADV_EXT=y
+CONFIG_BT_CTLR_ADV_PERIODIC=y
+
+# Enable Direction Finding TX Feature including AoA and AoD
+CONFIG_BT_CTLR_DF=y
+CONFIG_BT_CTLR_DF_ANT_SWITCH_RX=n
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf:9]: `RX feature`?
The disabled options should related to TX not RX.

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay:4]: `SPDX-License-Identifier: LicenseRef-Nordic-5-Clause`?

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+&radio {
+	status = "okay";
+	/* This is an example number of antennas that may be available
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay:9]: We are not going to use configuration for antenna matrix designed by Nordic here?

Comment (MarekPieta) [reply samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay:9]: Also Rx instead of Tx

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay (hunk 4 above)
Comment (MarekPieta) [reply samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay:9]: Also `antenna switching when AoA is enabled` ?

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst (hunk 1 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst]: Corrected, thank you.

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf (hunk 2 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf]: The whole file content is wrong. I've changed it to proper one.

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay (hunk 3 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay]: The whole file content is wrong. I've changed it to proper one.

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay (hunk 4 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.overlay]: The whole file content is wrong. I've changed it to proper one.

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf:2]: 2021 :)

**Review by KAGA164 — APPROVED** [review]

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+CONFIG_BT_CTLR=y
+CONFIG_BT_LL_SW_SPLIT=y
+
+CONFIG_BT_CTLR_ADV_EXT=y
+CONFIG_BT_CTLR_ADV_PERIODIC=y
```
Comment (MarekPieta) [inline samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf:11]: `CONFIG_BT_CTLR_SYNC_PERIODIC`?

Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf]: You are right, corrected. Thank you!

File: samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf (hunk 5 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52820.conf]: corrected

**Review by MarekPieta — APPROVED** [review]

**Comment by ppryga** [comment]
@mia-ko could you take a look on documentation here?

**Review by mia-ko — APPROVED** [review]

**Pull request merged**